Lib/ntpath.py

```diff
@@ -554,36 +554,48 @@ def normpath(path):
         return prefix + sep.join(comps)
 
 
-def _abspath_fallback(path):
-    """Return the absolute version of a path as a fallback function in case
-    `nt._getfullpathname` is not available or raises OSError. See bpo-31047 for
-    more.
-
-    """
-
-    path = os.fspath(path)
-    if not isabs(path):
-        if isinstance(path, bytes):
-            cwd = os.getcwdb()
-        else:
-            cwd = os.getcwd()
-        path = join(cwd, path)
-    return normpath(path)
-
 # Return an absolute path.
 try:
     from nt import _getfullpathname
 
 except ImportError: # not running on Windows - mock up something sensible
-    abspath = _abspath_fallback
+    def abspath(path):
+        """Return the absolute version of a path."""
+        path = os.fspath(path)
+        if not isabs(path):
+            if isinstance(path, bytes):
+                cwd = os.getcwdb()
+            else:
+                cwd = os.getcwd()
+            path = join(cwd, path)
+        return normpath(path)
 
 else:  # use native Windows method on Windows
     def abspath(path):
         """Return the absolute version of a path."""
         try:
             return _getfullpathname(normpath(path))
         except (OSError, ValueError):
-            return _abspath_fallback(path)
+            # See gh-75230, handle outside for cleaner traceback
+            pass
+        path = os.fspath(path)
+        if not isabs(path):
+            if isinstance(path, bytes):
+                sep = b'/'
```

Comment:
```suggestion
                sep = b'\\'
```

Reply:
This suggestion implies there is missing test coverage for this change.

Reply:
Actually not, the path is normalised afterwards. I was already starting to wonder why the test was passing on the previous pull request.

Reply:
They passed because the tests are either incomplete or incorrectly written.

Reply:
No, the test was complete, it passed because `normpath("\x00:/spam") == "\x00:\\spam"`.

Reply:
I believe this is everything I can test here. If you agree you can resolve both conversations.

Reply:
My suggestion wasn't because using "/" would return the wrong result. It's just fundamentally wrong to define `sep = '/'` in this context, in which that's the definition of `altsep`. The subsequent `normpath()` call was fixing the mistake, but there was no reason to not use the correct character from the start.

Reply:
I agree. It remains confusing that Windows needs "\\" instead of "/".